MicroCabal confuses ghc-boot and ghc-boot-th when searching for the version of ghc-boot
--- a/MicroCabal/src/MicroCabal/Backend/MHS.hs
+++ b/MicroCabal/src/MicroCabal/Backend/MHS.hs
@@ -76,7 +76,7 @@
   dir <- getMhsDir env
   pkgs <- listDirectory (dir </> "packages")
   let n = pkgName ++ "-"
-  case [ readVersion vers | p <- pkgs, Just verspkg <- [stripPrefix n p], Just vers <- [stripSuffix ".pkg" verspkg] ] of
+  case [ readVersion vers | p <- pkgs, Just verspkg <- [stripPrefix n p], Just vers <- [stripSuffix ".pkg" verspkg], all (`elem`("0123456789."::String)) vers ] of
     [v] -> return v
     []  -> error $ "Not installed: " ++ pkgName
     _   -> error $ "Multiple version: " ++ pkgName
